src/pendaftaran-agen/pendaftaran-agen.service.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  ForbiddenException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common'
+import { RepositoryService } from 'src/repository/repository.service'
+import { GetPendaftaranAgenQueryParamDTO } from './pendaftaran-agen.DTO'
+import { MailService } from 'src/mail/mail.service'
+
+@Injectable()
+export class PendaftaranAgenService {
+  constructor(
+    private readonly repository: RepositoryService,
+    private readonly mail: MailService
+  ) {}
+
+  async getAllPendaftaranAgen({
+    email,
+    nama,
+    status,
+  }: GetPendaftaranAgenQueryParamDTO) {
+    const pendaftaranAgen = await this.repository.pendaftaranAgen.findAll(
+      email,
+      nama,
+      status
+    )
+
+    return { pendaftaranAgen }
+  }
+
+  async getPendaftaranAgenById(id: string) {
+    const pendaftaranAgen = await this.getPendaftaranFromRepo(id)
+    const { password: _, ...res } = pendaftaranAgen
+
+    return { pendaftaranAgen: res }
+  }
+
+  async acceptPendaftaranAgen(id: string) {
+    const pendaftaranAgen = await this.getPendaftaranFromRepo(id)
+
+    const { email, nama, password } = pendaftaranAgen
+    const user = await this.repository.pengguna.findByEmail(email)
+
+    if (!!user) {
+      throw new ForbiddenException('User with the given email already exists')
+    }
+
+    if (pendaftaranAgen.statusPendaftaran !== 'DIAJUKAN') {
+      throw new ForbiddenException('PendaftaranAgen has already been processed')
```

Comment:
kenapa forbidden?

Reply:
forbidden lebih ke arah permission to access the resource g sih? gua mikirnya better dipakaiin bad req

Reply:
This status is similar to 401, but for the 403 Forbidden status code, re-authenticating makes no difference. The access is tied to the application logic, such as insufficient rights to a resource.

Tapi kalau dari penjelasan itu, lebih ke masalah akses g sih?

Reply:
hmmm okeh, sepertinya aku setuju jg dengan lu bon